File: Projects/0_Message_Receiver/go/tcp_receiver.go
package main

import (
	"fmt"
	"net"
	"os"
	"strings"
)

func receiver() error {
	// Create a TCP listener on localhost:1369
	listener, err := net.Listen("tcp", ":1369")
	if err != nil {
		return err
	}
	// Accept incoming connection
	conn, err := listener.Accept()
	if err != nil {
		return err
	}
	defer conn.Close()

	fmt.Println("Listening for messages...")
	for {
		dataBuff := make([]byte, 1024)
		// Read any incoming messages and print them
		_, err := conn.Read(dataBuff)
		if err != nil {
			return err
		}

		message := strings.TrimSpace(string(dataBuff))
		fmt.Println(message)
		if message == "exit" {
			break
		}
	}

	return nil
}

func sender() error {
	// Create a connection and dial localhost:1369
	conn, err := net.Dial("tcp", "127.0.0.1:1369")
	if err != nil {
		return err
	}
	defer conn.Close()

	fmt.Println("Connection successful")
	var message []byte
	for {
		fmt.Print("Insert message to send (send `exit` to stop): ")
		fmt.Scan(&message)
		// Write input message to listener
		_, err := conn.Write(message)
		if err != nil {
			return err
		}

		if string(message) == "exit" {
			break
		}
	}

	return nil
}

func main() {
	var option string
	fmt.Println("[send] or [receive] data?")
	fmt.Scan(&option)
	switch option {
	case "send":
		err := sender()
		if err != nil {
			fmt.Println(err)
		}
	case "receive":
		err := receiver()
		if err != nil {
			fmt.Println(err)
		}
	default:
		os.Exit(0)
	}
}

File: Projects/0_Message_Receiver/python/tcp_receiver.py
import socket

def receiver():
    host = ''
    port = 1369
    # Open socket in TCP mode and bind it to localhost:1369
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sck:
        sck.bind((host, port))
        # Listen and accept connections
        sck.listen(1)
        conn, address = sck.accept()

        data = None
        print('Listening for messages...')
        while data != b'exit':
            # Print any data received
            data = conn.recv(1024)
            print('>', data.decode())


def sender():
    localhost = '127.0.0.1'
    port = 1369
    # Open socket in TCP and connect to localhost:1369
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sck:
        sck.connect((localhost, port))
        print('Connection successful')

        message = None
        while message != 'exit':
            # Send input data
            message = input('Insert message to send (send `exit` to stop): ')
            sck.sendall(message.encode('utf-8'))


if __name__ == '__main__':
    option = input('[send] or [receive] data? ')
    match option:
        case 'send':
            sender()
        case 'receive':
            receiver()
        case _:
            exit()